kartothek/io_components/metapartition.py

```diff
@@ -1549,6 +1572,58 @@ def delete_from_store(self, dataset_uuid, store):
             store.delete(file_key)
         return self.copy(files={}, data={}, metadata={})
 
+    def get_parquet_metadata(self, store, table_name):
+        """
+        Retrieve the parquet metadata for the MetaPartition.
+        Especially relevant for calculating dataset statistics.
+
+        Parameters
+        ----------
+        store
+          A factory function providing a KeyValueStore
+        table_name
+          Name of the kartothek table for which the statistics should be retrieved
+
+        Returns
+        -------
+        pd.DataFrame
+          A DataFrame with relevant parquet metadata
+        """
+        if not isinstance(table_name, str):
+            raise TypeError("Expecting a string for parameter `table_name`.")
+
+        if callable(store):
+            store = store()
+
+        if table_name in self.files:
+            with store.open(self.files[table_name]) as fd:
+                pq_metadata = pa.parquet.ParquetFile(fd).metadata
+            data = {
+                "partition_label": self.label,
+                "number_rows_total": pq_metadata.num_rows,
+                "number_row_groups": pq_metadata.num_row_groups,
+                "serialized_size": pq_metadata.serialized_size,
+            }
+            data["row_group_id"] = range(data["number_row_groups"])
+
+            data["row_group_byte_size"], data["number_rows_per_row_group"] = zip(
+                *[
+                    (
+                        pq_metadata.row_group(rg_id).total_byte_size,
+                        pq_metadata.row_group(rg_id).num_rows,
+                    )
+                    for rg_id in data["row_group_id"]
+                ]
+            )
```

Comment:
tbh, I would prefer this to be expressed a bit simpler. At least for me it is very hard to read and understand what a "zip of list of tuples" actually generates. Why not

```suggestion
            data["row_group_byte_size"] = []
            data["number_rows_per_row_group"] = []

            for rg_id in data["row_group_id"]:
                data["row_group_byte_size"].append(
                    pq_metadata.row_group(rg_id).total_byte_size
                )
                data["number_rows_per_row_group"].append(
                    pq_metadata.row_group(rg_id).num_rows
                )
```

That's easier to read and probably generates a smaller diff when changing, i.e. easier to maintain